src/lib/structures/Command.ts

```diff
@@ -0,0 +1,214 @@
+import { AliasPiece, AliasPieceOptions, Permissions, PermissionsResolvable } from '@klasa/core';
+
+import type { CommandStore } from './CommandStore';
+import type { ChannelType } from '@klasa/dapi-types';
+
+import { RateLimitManager } from '@klasa/ratelimits';
+import { mergeDefault } from '@klasa/utils';
+
+import { commandDefaults } from '../util/constants';
+import { CooldownLevel } from '../types/Enums';
+
+export abstract class Command extends AliasPiece {
+	/**
+	 * Permissions required by the bot to run the command
+	 * @since 0.0.1
+	 */
+	public requiredPermissions: Permissions;
+
+	/**
+	 * Delete command's response if the trigger message was deleted
+	 * @since 0.0.1
+	 */
+	public deletable: boolean;
```

Comment:
this can use a better name

Reply:
I thought about but couldn't think of a better name.